src/appsync.ts

```diff
@@ -190,25 +191,87 @@ export class AppsyncResolver<
                     api,
-                    service.resource.node.addr,
-                    {
-                      api,
-                      table: service.resource,
-                    }
-                  )
-                : new appsync.LambdaDataSource(
+                    table: service.resource,
+                  }
+                );
+              } else if (service.kind === "Function") {
+                return new appsync.LambdaDataSource(
+                  api,
+                  service.resource.node.addr,
+                  {
                     api,
-                    service.resource.node.addr,
-                    {
-                      api,
-                      lambdaFunction: service.resource,
-                    }
-                  )
-            );
+                    lambdaFunction: service.resource,
+                  }
+                );
+              } else if (service.kind === "StepFunction") {
+                const ds = new appsync.HttpDataSource(
+                  api,
+                  service.resource.node.addr,
+                  {
+                    api,
+                    endpoint: `https://${
+                      service.getStepFunctionType() ===
+                      aws_stepfunctions.StateMachineType.EXPRESS
+                        ? "sync-states"
+                        : "states"
+                    }.${service.resource.stack.region}.amazonaws.com/`,
+                    authorizationConfig: {
+                      signingRegion: api.stack.region,
+                      signingServiceName: "states",
+                    },
+                  }
+                );
+                if (
+                  service.getStepFunctionType() ===
+                  aws_stepfunctions.StateMachineType.EXPRESS
+                ) {
+                  service.grantStartSyncExecution(ds.grantPrincipal);
+                } else {
+                  service.grantStartExecution(ds.grantPrincipal);
+                }
+                return ds;
+              } else {
+                // TODO: use HTTP resolvers for the AWS-SDK types.
+                throw new Error(
+                  `${service.kind} cannot be used within an Appsync Resolver, please use an Appsync-compatible function`
+                );
+              }
+            });
+
+            let returnValName = "$context.result";
+            let pre: string | undefined;
+            if (service.kind === "StepFunction") {
+              returnValName = "$sfn__result";
+
+              if (
+                service.getStepFunctionType() ===
+                aws_stepfunctions.StateMachineType.EXPRESS
+              ) {
+                pre = `#if($context.result.statusCode == 200)
+  #set(${returnValName} = $util.parseJson($context.result.body))
+  #if(${returnValName}.output == 'null')
+    $util.qr(${returnValName}.put("output", $null))
+  #else
+    #set(${returnValName}.output = $util.parseJson(${returnValName}.output))
```

Comment:
Haven't thought about it. Do you have suggestions? I haven't experienced that aspect of the service.

Reply:
Looks like this may not be an issue for app sync. 
https://docs.aws.amazon.com/step-functions/latest/apireference/API_HistoryEventExecutionDataDetails.html

I couldn't find any docs that reference it at least. I feel like it was a concern with step function to step functions, but I guess we'll find out as we go.